src/SerializableClosure.php

```diff
@@ -115,7 +115,12 @@ public function unserialize($serialized)
     {
         // Unserialize the data and reconstruct the SuperClosure.
         $this->data = unserialize($serialized);
-        $this->reconstructClosure();
+        try {
+            $this->reconstructClosure();
+        } catch (\ParseException $e) {
+            // Discard the parse exception, we'll throw a custom one
```

Comment:
`ClosureUnserializationException` is a custom exception already. @nikic's change just makes it work the same way in PHP7 as in PHP5, because PHP7 now has a `ParseException` (via [RFC: Exceptions in the engine](https://wiki.php.net/rfc/engine_exceptions_for_php7)).